lightning/src/ln/channel.rs

```diff
@@ -3678,6 +3678,29 @@ impl<Signer: Sign> Channel<Signer> {
 		Ok((None, timed_out_htlcs))
 	}
 
+	/// Indicates the funding transaction is no longer confirmed in the main chain. This may
+	/// force-close the channel, but may also indicate a harmless reorganization of a block or two
+	/// before the channel has reached funding_locked and we can just wait for more blocks.
+	pub fn funding_transaction_unconfirmed(&mut self) -> Result<(), msgs::ErrorMessage> {
+		if self.funding_tx_confirmation_height != 0 {
+			// We handle the funding disconnection by calling update_best_block with a height one
+			// below where our funding was connected, implying a reorg back to conf_height - 1.
+			let reorg_height = self.funding_tx_confirmation_height - 1;
+			let best_time = self.update_time_counter;
+			match self.update_best_block(reorg_height, best_time) {
+				Ok((funding_locked, timed_out_htlcs)) => {
+					assert!(funding_locked.is_none(), "We can't generate a funding with 0 confirmations?");
+					assert!(timed_out_htlcs.is_empty(), "We can't have accepted HTLCs with a timeout before our funding confirmation?");
+					Ok(())
+				},
+				Err(e) => Err(e)
+			}
+		} else {
+			// We never learned about the funding confirmation anyway, just ignore
+			Ok(())
```

Comment:
Is this theoretically supposed to be unreachable, but it seems harmless so we can just `Ok(())` in practice?

Reply:
Yea, I think it should be really rare, but you could see some async process getting delayed and ending up calling `transaction_unconfirmed` twice in a row which could hit this.